core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -1970,6 +1970,11 @@ wasmtime_ssp_path_open(wasm_exec_env_t exec_env, struct fd_table *curfds,
     // Convert file descriptor flags.
     if ((fs_flags & __WASI_FDFLAG_APPEND) != 0)
         noflags |= O_APPEND;
+    if ((fs_flags & __WASI_FDFLAG_NONBLOCK) != 0)
+        noflags |= O_NONBLOCK;
+    if (write && (noflags & (O_APPEND | O_TRUNC)) == 0)
+        needed_inheriting |= __WASI_RIGHT_FD_SEEK;
```

Comment:
is this related to the rest of the PR?

Reply:
Yes, this is just moving the existing code around (it made sticking all the sync flags behind an ifdef easier). If you look a few lines below here, the same lines are deleted.